arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1380,6 +1470,37 @@ void ClusterInfo::loadCurrent() {
 
     DatabaseCollectionsCurrent databaseCollections;
 
+    auto const existingCollections = newCollections.find(databaseName);
+    if (existingCollections != newCollections.end()) {
+      auto const& nc = newCurrent.find(databaseName);
+      if (nc != newCurrent.end()) {
+        auto ncs = nc->second->slice()[0];
+        std::vector<std::string> path{
+          AgencyCommHelper::path(), "Current", "Collections", databaseName};
+        for (auto const& ec : existingCollections->second) {
+          auto const& cid = ec.first;
+          path.push_back(cid);
+          if (ncs.hasKey(path)) {
+            std::shared_ptr<VPackBuilder> cur;
+            {
+              READ_LOCKER(guard, _currentProt.lock);
+              cur = _current.find(databaseName)->second;
```

Comment:
what if `_current.find(databaseName)` returns `_current.end()` here?

Reply:
cannot happen. newCurrent was copied from it and it has not changed since.

Reply:
Got it. All databases from `changeSet` (except the empty string one) are also in `newCurrent`, and the empty string one is currently handled with a separate if clause.
I agree it should work.